compiler/utils/btrees.nim

```diff
@@ -16,8 +16,7 @@ const
   Mhalf = M div 2
 
 type
-  # TODO: This type should be marked acyclic
-  Node[Key, Val]  = ref object
+  Node[Key, Val] {.acyclic.} = ref object
```

Comment:
Same thing

Reply:
It's a node in a balanced *tree*. I figured that no cycles being possible in a node tree is self-explanatory.